### PR TITLE
add image stream and WorkflowTemplate for thoth-messaging

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies
We need to be able to use the messaging `CLI` as a step in workflows.

### CODE REVIEW

**Comment by harshad16** [comment]
ah, which component would be triggering messaging ?

File: messaging/base/argo-workflows/send-message.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (pacospace) [inline messaging/base/argo-workflows/send-message.yaml:1]: I think this can go to core actually as many workflows can use it? @goern @harshad16

Comment (harshad16) [reply messaging/base/argo-workflows/send-message.yaml:1]: yeah, I agree, if this is not a standalone component. if this triggered by some other component then this should go in that application subdirectory.

Comment (pacospace) [reply messaging/base/argo-workflows/send-message.yaml:1]: It can be used as workflow-helpers in different components basically :)

Comment (KPostOffice) [reply messaging/base/argo-workflows/send-message.yaml:1]: Should I also add a sub-directory `argo-workflows` in `core/base`? To put the WorkflowTemplate in?

Comment (harshad16) [reply messaging/base/argo-workflows/send-message.yaml:1]: yeah that would be appropriate space to add these

**Comment by pacospace** [comment]
> ah, which component would be triggering messaging ?

Solver will use it and Adviser also when we will migrate the current task

File: core/base/argo-workflows/send-message.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            value: "{{inputs.parameters.THOTH_MESSAGING_CREATE_IF_NOT_EXISTS}}"
+          - name: THOTH_MESSAGING_MESSAGE_CONTENTS
+            value: "{{inputs.parameters.THOTH_MESSAGING_MESSAGE_CONTENTS}}"
+          - name: THOTH_MESSAGING_PARTITIONS
+            value: "{{inputs.parameters.THOTH_MESSAGING_PARTITIONS}}"
+          - name: KAFKA_BOOTSTRAP_SERVERS
+            valueFrom:
+              configMapKeyRef:
+                name: KAFKA_BOOTSTRAP_SERVERS
+                key: kafka-bootstrap-servers
```
Comment (pacospace) [inline core/base/argo-workflows/send-message.yaml:36]: `kafka`

Comment (pacospace) [reply core/base/argo-workflows/send-message.yaml:36]:
```
          - name: KAFKA_BOOTSTRAP_SERVERS
            valueFrom:
              configMapKeyRef:
                key: kafka-bootstrap-servers
                name: kafka
```

**Pull request merged**